src/libraries/System.Text.Json/tests/Serialization/CustomConverterTests/CustomConverter.Dynamic.Sample.Tests.cs

```diff
@@ -0,0 +1,315 @@
+            child.ChildProp = 1;
+            obj.Child = child;
+
+            string json = JsonSerializer.Serialize(obj, options);
+            Assert.Equal(ExpectedJson, json);
+        }
+
+        [Fact]
+        public static void VerifyMutableDom_WithIndexers()
+        {
+            const string ExpectedJson = "{\"MyString\":\"Hello!\",\"MyNull\":null,\"MyBoolean\":false,\"MyArray\":[\"2\"]," +
+                "\"MyInt\":43,\"MyDouble\":4.3,\"MyDateTime\":\"2020-07-08T00:00:00\",\"MyGuid\":\"ed957609-cdfe-412f-88c1-02daca1b4f51\"," +
+                "\"MyObject\":{\"MyString\":\"Hello!!\"},\"Child\":{\"ChildProp\":1}}";
+
+            var options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+
+            JsonDynamicObject obj = (JsonDynamicObject)JsonSerializer.Deserialize<object>(DynamicTests.Json, options);
+
+            // Change some primitives
+            obj["MyString"] = "Hello!";
+            obj["MyBoolean"] = false;
+            obj["MyInt"] = 43;
+            obj["MyDouble"] = 4.3;
+
+            ((IDictionary<string, object>)obj["MyObject"])["MyString"] = "Hello!!";
+
+            // Use Dictionary for new child property
+            var child = new Dictionary<string, object>();
+            child["ChildProp"] = 1;
+            obj["Child"] = child;
+
+            ((IList<object>)obj["MyArray"]).Clear();
+            ((IList<object>)obj["MyArray"]).Add("1");
+            ((IList<object>)obj["MyArray"])[0]="2"; // replace the "1"
+
+            string json = JsonSerializer.Serialize(obj, options);
+            Assert.Equal(ExpectedJson, json);
+        }
+
+        [Fact]
+        public static void DynamicObject_MissingProperty()
+        {
+            var options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+            dynamic obj = JsonSerializer.Deserialize<dynamic>("{}", options);
+
+            // We return null here; ExpandoObject throws for missing properties.
+            Assert.Equal(null, obj.NonExistingProperty);
+        }
+
+        [Fact]
+        public static void DynamicObject_CaseSensitivity()
+        {
+            var options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+            dynamic obj = JsonSerializer.Deserialize<dynamic>("{\"MyProperty\":42}", options);
+
+            Assert.Equal(42, (int)obj.MyProperty);
+            Assert.Null(obj.myproperty);
+            Assert.Null(obj.MYPROPERTY);
+
+            options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+            options.PropertyNameCaseInsensitive = true;
+            obj = JsonSerializer.Deserialize<dynamic>("{\"MyProperty\":42}", options);
+
+            Assert.Equal(42, (int)obj.MyProperty);
+            Assert.Equal(42, (int)obj.myproperty);
+            Assert.Equal(42, (int)obj.MYPROPERTY);
+        }
+
+        [Fact]
+        public static void NamingPoliciesAreNotUsed()
```

Comment:
Yes. Since the properties in JSON directly correspond to the CLR property names (though `dynamic`), applying a naming policy doesn't make sense since the JSON wouldn't be able to round-trip. It would be possible to use a naming policy, however,  through a two-way naming policy or preserving the original property names, but I don't think that is expected or desired.